### PR TITLE
Update FFC messaging module to v2

### PR DESCRIPTION
**Pull request by paulsimonandrews**

### CODE REVIEW

File: app/messaging/agreement-changed-v1.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+    const totalPayment = calculatePayment(body)
 
-  body.totalPayment = totalPayment
-  await updateAgreement({ body, correlationId })
+    body.totalPayment = totalPayment
+    await updateAgreement({ body, correlationId })
+
+    await calculatorReceiver.completeMessage(msg)
+  } catch (err) {
+    console.error('Unable to process message:', err)
```
Comment (st3v3nhunt) [inline app/messaging/agreement-changed-v1.js]: Can this be replaced with `logger.logError`?

File: app/messaging/determine-agreement-version.js [hunk 2]
```diff
@@ -2,13 +2,16 @@ const v1 = require('./agreement-changed-v1')
 const v2 = require('./agreement-changed-v2')
 const { log } = require('../services/logger')
 
-module.exports = async function (msg) {
+module.exports = async function (msg, calculatorReceiver) {
   // NOTE: not particularly robust for determining version but ok for now
+
+  console.log('GOT MESSAGE!')
```
Comment (st3v3nhunt) [inline app/messaging/determine-agreement-version.js]: Is this required? If so, can it be updated to use `logger.log`?

File: app/messaging/agreement-changed-v2.js [hunk 3]
```diff
@@ -1,31 +1,38 @@
...
+    }
+    await updateAgreement({ body, correlationId })
 
-  body.payments = {
-    actionsPayment,
-    standardsPayment,
-    totalPayment: actionsPayment + standardsPayment
+    await calculatorReceiver.completeMessage(msg)
+  } catch (err) {
+    console.error('Unable to process message:', err)
```
Comment (st3v3nhunt) [inline app/messaging/agreement-changed-v2.js]: Can this be replaced with `logger.logError`?

**Review by st3v3nhunt — CHANGES_REQUESTED** [review]
A few `console` statements that could probably be replaced with the logger.

On placement of error handling for messages that are not processed, could this be moved into the `updateAgreement` function so the abandon message only needs to be handled once.

File: app/messaging/determine-agreement-version.js (hunk 2 above)
Comment (paulsimonandrews) [reply app/messaging/determine-agreement-version.js]: Oops, debugging statement left in. Gone now.

**Review by st3v3nhunt — APPROVED** [review]
LGTM 👍

**Pull request merged**